### ISSUE TITLE
Salt-SSH in latest develop fails to run due to missing msgpack or msgpack_pure modules on client

### ISSUE THREAD

**Issue by Xiol** [saltstack/salt#31822]
### Description of Issue/Question

As far as I can see, Salt-SSH is failing to execute on a remote host because the remote host does not have msgpack or msgpack_pure Python modules installed. Looking at previous bugs of this nature I'm assuming this is not intended.

### Setup

We're executing Salt-SSH like so:

```/usr/bin/salt-ssh --out json --no-color -E '(SID41929|SID41932)' --no-host-keys --static -l quiet --log-file /var/log/ed/salt-logs/1457713116_SID41929_SID41932 --log-file-level trace --roster-file /var/lib/ed/salt/roster -c /var/lib/ed/salt state.sls common_centos,common_centos7 pillar='{...}'```

FWIW, we execute this using another application and capture the JSON output for processing, hence the weird nature of our execution call.

### Steps to Reproduce Issue

Attempt to use Salt-SSH against a CentOS 7 client without msgpack installed:

```
2016-03-11 16:18:43,285 [salt.utils.vt.Terminal.PID-23788.STDERR][TRACE   ][23769] SALT_ARGV: ['/usr/bin/python2.7', '/tmp/.root_e0982d_salt/salt-call', '--local', '--metadata', '--out', 'json', '-l', 'quiet', '-
c', '/tmp/.root_e0982d_salt', '--', 'test.opts_pkg']
_edbc7885e4f9aac9b83b35999b68d015148caf467b78fa39c05f669c0ff89878
2016-03-11 16:18:43,285 [salt.utils.vt.Terminal.PID-23788.STDOUT][TRACE   ][23769] _edbc7885e4f9aac9b83b35999b68d015148caf467b78fa39c05f669c0ff89878
2016-03-11 16:18:43,345 [salt.utils.vt.Terminal.PID-23791.STDERR][TRACE   ][23767] SALT_ARGV: ['/usr/bin/python2.7', '/tmp/.root_e0982d_salt/salt-call', '--local', '--metadata', '--out', 'json', '-l', 'quiet', '-
c', '/tmp/.root_e0982d_salt', '--', 'test.opts_pkg']
_edbc7885e4f9aac9b83b35999b68d015148caf467b78fa39c05f669c0ff89878
2016-03-11 16:18:43,345 [salt.utils.vt.Terminal.PID-23791.STDOUT][TRACE   ][23767] _edbc7885e4f9aac9b83b35999b68d015148caf467b78fa39c05f669c0ff89878
2016-03-11 16:18:43,579 [salt.utils.vt.Terminal.PID-23788.STDERR][TRACE   ][23769] [CRITICAL] Unable to import msgpack or msgpack_pure python modules
2016-03-11 16:18:43,589 [salt.utils.vt.Terminal.PID-23788.STDERR][TRACE   ][23769] [ERROR   ] An un-handled exception was caught by salt's global exception handler:
ImportError: No module named msgpack
Traceback (most recent call last):
  File "/tmp/.root_e0982d_salt/salt-call", line 13, in <module>
    from salt.scripts import salt_call
  File "/tmp/.root_e0982d_salt/py2/salt/scripts.py", line 17, in <module>
    from salt import cloud, defaults
  File "/tmp/.root_e0982d_salt/py2/salt/cloud/__init__.py", line 29, in <module>
    import salt.config
  File "/tmp/.root_e0982d_salt/py2/salt/config/__init__.py", line 41, in <module>
    import salt.utils.sdb
  File "/tmp/.root_e0982d_salt/py2/salt/utils/sdb.py", line 9, in <module>
    import salt.loader
  File "/tmp/.root_e0982d_salt/py2/salt/loader.py", line 29, in <module>
    import salt.utils.event
  File "/tmp/.root_e0982d_salt/py2/salt/utils/event.py", line 82, in <module>
    import salt.transport.ipc
  File "/tmp/.root_e0982d_salt/py2/salt/transport/ipc.py", line 10, in <module>
    import msgpack
ImportError: No mod
2016-03-11 16:18:43,600 [salt.utils.vt.Terminal.PID-23788.STDERR][TRACE   ][23769] ule named msgpack
Traceback (most recent call last):
  File "/tmp/.root_e0982d_salt/salt-call", line 13, in <module>
    from salt.scripts import salt_call
  File "/tmp/.root_e0982d_salt/py2/salt/scripts.py", line 17, in <module>
    from salt import cloud, defaults
  File "/tmp/.root_e0982d_salt/py2/salt/cloud/__init__.py", line 29, in <module>
    import salt.config
  File "/tmp/.root_e0982d_salt/py2/salt/config/__init__.py", line 41, in <module>
    import salt.utils.sdb
  File "/tmp/.root_e0982d_salt/py2/salt/utils/sdb.py", line 9, in <module>
    import salt.loader
  File "/tmp/.root_e0982d_salt/py2/salt/loader.py", line 29, in <module>
    import salt.utils.event
  File "/tmp/.root_e0982d_salt/py2/salt/utils/event.py", line 82, in <module>
    import salt.transport.ipc
  File "/tmp/.root_e0982d_salt/py2/salt/transport/ipc.py", line 10, in <module>
    import msgpack
ImportError: No module named msgpack
```

### Versions Report

```
Salt Version:
           Salt: 2016.3.0-195-g4147ff2
 
Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
           Mako: 0.9.1
         PyYAML: 3.10
          PyZMQ: 14.0.1
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
           RAET: Not Installed
        Tornado: 4.3
            ZMQ: 4.0.4
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.3.0
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed
 
System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-32-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

**Comment by jfindlay** [comment]
@Xiol, thanks for reporting.  I can reproduce this
```py
root@jmoney-ubuntu-14:~/salt# salt-ssh jmoney-centos-7 -i test.ping
[WARNING ] Key master_tops with value {} has an invalid type of dict, a bool is required for this value
jmoney-centos-7:
    ----------
    retcode:
        1
    stderr:
        [CRITICAL] Unable to import msgpack or msgpack_pure python modules
        [ERROR   ] An un-handled exception was caught by salt's global exception handler:
        ImportError: No module named msgpack
        Traceback (most recent call last):
          File "/tmp/.root_aa4167_salt/salt-call", line 13, in <module>
            from salt.scripts import salt_call
          File "/tmp/.root_aa4167_salt/py2/salt/scripts.py", line 17, in <module>
            from salt import cloud, defaults
          File "/tmp/.root_aa4167_salt/py2/salt/cloud/__init__.py", line 29, in <module>
            import salt.config
          File "/tmp/.root_aa4167_salt/py2/salt/config/__init__.py", line 41, in <module>
            import salt.utils.sdb
          File "/tmp/.root_aa4167_salt/py2/salt/utils/sdb.py", line 9, in <module>
            import salt.loader
          File "/tmp/.root_aa4167_salt/py2/salt/loader.py", line 29, in <module>
            import salt.utils.event
          File "/tmp/.root_aa4167_salt/py2/salt/utils/event.py", line 82, in <module>
            import salt.transport.ipc
          File "/tmp/.root_aa4167_salt/py2/salt/transport/ipc.py", line 10, in <module>
            import msgpack
        ImportError: No module named msgpack
        Traceback (most recent call last):
          File "/tmp/.root_aa4167_salt/salt-call", line 13, in <module>
            from salt.scripts import salt_call
          File "/tmp/.root_aa4167_salt/py2/salt/scripts.py", line 17, in <module>
            from salt import cloud, defaults
          File "/tmp/.root_aa4167_salt/py2/salt/cloud/__init__.py", line 29, in <module>
            import salt.config
          File "/tmp/.root_aa4167_salt/py2/salt/config/__init__.py", line 41, in <module>
            import salt.utils.sdb
          File "/tmp/.root_aa4167_salt/py2/salt/utils/sdb.py", line 9, in <module>
            import salt.loader
          File "/tmp/.root_aa4167_salt/py2/salt/loader.py", line 29, in <module>
            import salt.utils.event
          File "/tmp/.root_aa4167_salt/py2/salt/utils/event.py", line 82, in <module>
            import salt.transport.ipc
          File "/tmp/.root_aa4167_salt/py2/salt/transport/ipc.py", line 10, in <module>
            import msgpack
        ImportError: No module named msgpack
    stdout:
```
While trying to reproduce this on CentOS 5 (I misread your comment at first) I encountered another error.
```py
root@jmoney-ubuntu-14:~/salt# salt-ssh jmoney-centos-5 -i test.ping
[WARNING ] Key master_tops with value {} has an invalid type of dict, a bool is required for this value
jmoney-centos-5:
    ----------
    retcode:
        1
    stderr:
        Traceback (most recent call last):
          File "/tmp/.root_aa4167_salt/salt-call", line 13, in <module>
            from salt.scripts import salt_call
          File "/tmp/.root_aa4167_salt/py2/salt/scripts.py", line 17, in <module>
            from salt import cloud, defaults
          File "/tmp/.root_aa4167_salt/py2/salt/cloud/__init__.py", line 29, in <module>
            import salt.config
          File "/tmp/.root_aa4167_salt/py2/salt/config/__init__.py", line 41, in <module>
            import salt.utils.sdb
          File "/tmp/.root_aa4167_salt/py2/salt/utils/sdb.py", line 9, in <module>
            import salt.loader
          File "/tmp/.root_aa4167_salt/py2/salt/loader.py", line 29, in <module>
            import salt.utils.event
          File "/tmp/.root_aa4167_salt/py2/salt/utils/event.py", line 68, in <module>
            import tornado.iostream
          File "/tmp/.root_aa4167_salt/py2/tornado/iostream.py", line 40, in <module>
            from tornado.netutil import ssl_wrap_socket, ssl_match_hostname, SSLCertificateError, _client_ssl_defaults, _server_ssl_defaults
          File "/tmp/.root_aa4167_salt/py2/tornado/netutil.py", line 49, in <module>
            import backports.ssl_match_hostname
        ImportError: No module named backports.ssl_match_hostname
    stdout:
```
```yaml
root@jmoney-ubuntu-14:~/salt# salt-ssh --versions
Salt Version:
           Salt: 2016.3.0-n/a-5bcd923

Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: Not Installed
           Mako: 0.9.1
         PyYAML: 3.10
          PyZMQ: 14.0.1
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.4
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.8.2
        timelib: Not Installed

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-79-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

**Comment by jcftang** [comment]
I also notice this problem as well, my targeted minion is a debian jessie box (on arm)

```
salt-ssh cubox cmd.run uptime
cubox:
    ----------
    retcode:
        1
    stderr:
        [CRITICAL] Unable to import msgpack or msgpack_pure python modules
        [ERROR   ] An un-handled exception was caught by salt's global exception handler:
        ImportError: No module named msgpack
        Traceback (most recent call last):
          File "/tmp/.root_f6a67b_salt/salt-call", line 15, in <module>
            salt_call()
          File "/tmp/.root_f6a67b_salt/py2/salt/scripts.py", line 327, in salt_call
            import salt.cli.call
          File "/tmp/.root_f6a67b_salt/py2/salt/cli/call.py", line 6, in <module>
            from salt.utils import parsers
          File "/tmp/.root_f6a67b_salt/py2/salt/utils/parsers.py", line 28, in <module>
            import salt.config as config
          File "/tmp/.root_f6a67b_salt/py2/salt/config/__init__.py", line 41, in <module>
            import salt.utils.sdb
          File "/tmp/.root_f6a67b_salt/py2/salt/utils/sdb.py", line 9, in <module>
            import salt.loader
          File "/tmp/.root_f6a67b_salt/py2/salt/loader.py", line 29, in <module>
            import salt.utils.event
          File "/tmp/.root_f6a67b_salt/py2/salt/utils/event.py", line 82, in <module>
            import salt.transport.ipc
          File "/tmp/.root_f6a67b_salt/py2/salt/transport/ipc.py", line 10, in <module>
            import msgpack
        ImportError: No module named msgpack
        Traceback (most recent call last):
          File "/tmp/.root_f6a67b_salt/salt-call", line 15, in <module>
            salt_call()
          File "/tmp/.root_f6a67b_salt/py2/salt/scripts.py", line 327, in salt_call
            import salt.cli.call
          File "/tmp/.root_f6a67b_salt/py2/salt/cli/call.py", line 6, in <module>
            from salt.utils import parsers
          File "/tmp/.root_f6a67b_salt/py2/salt/utils/parsers.py", line 28, in <module>
            import salt.config as config
          File "/tmp/.root_f6a67b_salt/py2/salt/config/__init__.py", line 41, in <module>
            import salt.utils.sdb
          File "/tmp/.root_f6a67b_salt/py2/salt/utils/sdb.py", line 9, in <module>
            import salt.loader
          File "/tmp/.root_f6a67b_salt/py2/salt/loader.py", line 29, in <module>
            import salt.utils.event
          File "/tmp/.root_f6a67b_salt/py2/salt/utils/event.py", line 82, in <module>
            import salt.transport.ipc
          File "/tmp/.root_f6a67b_salt/py2/salt/transport/ipc.py", line 10, in <module>
            import msgpack
        ImportError: No module named msgpack
    stdout:
```

and

```
Salt Version:
           Salt: 2016.3.0rc1-96-gfe86a3d

Dependency Versions:
         Jinja2: 2.8
       M2Crypto: Not Installed
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
         Python: 2.7.11 (default, Dec  5 2015, 14:44:53)
           RAET: Not Installed
        Tornado: 4.3
            ZMQ: 4.1.2
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed

System Versions:
           dist:
        machine: x86_64
        release: 15.3.0
         system: Darwin
        version: 10.11.3 x86_64
```

**Comment by Xiol** [comment]
FWIW I've just tried this with the latest develop and it now seems to work - not sure if anything has been committed or if my test setup is slightly different (which it is, but last time I couldn't get it to run on anything).

**Comment by jfindlay** [comment]
@Xiol, thanks for the confirmation.  Are you able to also try 2016.3.0rc2?

**Comment by cachedout** [comment]
This is definitely fixed in 2016.3.0.
